.github/workflows/pr-checks.yml

```
@@ -0,0 +1,239 @@
+        swift test --filter "ResourceManagerTests|DownloadPriorityTests"
+        echo "✅ Core tests completed"
+
+  essential-tests:
+    name: Essential Tests
+    runs-on: macos-latest
+    needs: quick-validation
+    
+    strategy:
+      matrix:
+        test-suite:
+          - name: "Priority & Storage"
```

YAMLlint (1.37.1)

[warning] wrong indentation: expected 8 but found 10

(indentation)